doc/introduction/optimizers.rst

```diff
@@ -6,7 +6,18 @@
 Optimizers
 ==========
 
-When using the standard NumPy interface, PennyLane offers some custom-made optimizers.
+Optimizers are objects which can be used to automatically update the parameters of a quantum 
+or hybrid machine learning model. The Optimizers you should use are dependent on your choice
+of classical interface (NumPy, PyTorch, and TensorFlow), and are available from different access
+points. 
+
+Regardless of their origin, all Optimizers provide the same core functionality, 
+and PennyLane is fully compatible with all of them. 
+
+NumPy Interface
+^^^^^^^^^^^^^^^
```

Comment:
Nice, I like that you separated it into sections! This helps avoid people using PyTorch from attempting to use the built in optimizers.

Side thought: for some of the specialty optimizers (such as QNG, maybe a future Rotoselect), would it be difficult to make them work for PyTorch and TF? I suppose how you do gradients is different enough that we might have to provide TorchQNG, TFQNG, etc.